File: language-support/js/daml-ledger/tests/arbitrary/get_transaction_by_event_id_request.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as jsc from 'jsverify';
import * as ledger from '../../src';

export const GetTransactionByEventIdRequest: jsc.Arbitrary<ledger.GetTransactionByEventIdRequest> =
    jsc.tuple([jsc.string, jsc.array(jsc.string)]).smap(
        ([eventId, requestingParties]) => ({
            eventId: eventId,
            requestingParties: requestingParties
        }),
        (request) =>
            [request.eventId, request.requestingParties]
    );

File: canton/community/ledger/ledger-api-core/src/main/scala/com/digitalasset/canton/ledger/api/messages/transaction/GetTransactionByEventIdRequest.scala
// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.digitalasset.canton.ledger.api.messages.transaction

import com.daml.lf.data.Ref.Party
import com.digitalasset.canton.ledger.api.domain.EventId

final case class GetTransactionByEventIdRequest(
    eventId: EventId,
    requestingParties: Set[Party],
)
